server/controllers/profile.js

```diff
@@ -3,72 +3,187 @@ const asyncHandler = require("express-async-handler");
 const Profile = require("../models/Profile");
 const User = require("../models/User");
 
+// @route POST /profile/:id
+// @descr Given a user ID and profile parameters, create a profile
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const userId = req.params.id;
+    const profileData = new Profile(req.body);
+
+    // Needs check to ensure that info sender owns the profile,
+    // and that there is no profile already there
+    try {
+        await profileData.save().then(() => {
+            User.updateOne(
+                { _id: userId },
+                { $set: { profile: profileData._id } },
+                (err) => {
+                    if (err) {
+                        console.log(err);
+                        res.send(err);
+                    } else {
+                        res.json({ success: true, msg: "Message added" });
+                    }
+                }
+            );
+        });
+    } catch (err) {
+        console.log(err);
+        res.send(err);
+    }
```

Comment:
Express-async-handler will deal with catching thrown errors meaning the try/catch is unnecessary here.